routers/repo/view.go

```diff
@@ -264,16 +264,37 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 // Home render repository home page
 func Home(ctx *context.Context) {
 	if len(ctx.Repo.Repository.Units) > 0 {
-		tp := ctx.Repo.Repository.Units[0].Type
-		if tp == models.UnitTypeCode {
-			renderCode(ctx)
-			return
+		var firstUnit struct {
```

Comment:
Using double pointers and internal structs seems unnecessarily complicated. Can we please just do something like the following:
```
if ctx.Repo.Repository.UnitEnabled(models.UnitTypeCode) {
	renderCode(ctx)
	return
}
redirectUnitType = ctx.Repo.Repository.Units[0].Type
for _, unit := range ctx.Repo.Repository.Units {
	// LessThan is a helper method to add to unit type, we can 
	// implement it so that it favors non-external units
	if unit.Type.LessThan(redirectUnitType) {
		redirectUnitType = unit.Type
	}
}
redirectUnit := models.Units[redirectUnitType]
ctx.Redirect(fmt.Sprintf("%s/%s%s", setting.AppSubURL,
	ctx.Repo.Repository.FullName(),
	redirectUnit.URI))
```

Reply:
OK, I'll use map version instead. The reason for not using only 1 variable is to avoid redirecting to external tracker or wiki if there is another (internal) tab available.